metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1585,6 +1642,27 @@ def _create_lineage_to_upstream_tables(
                 aspect=upstream_lineage,
             )
 
+    def _substitude_tableau_parameters(self, query: str) -> str:
+        if not query:
+            return query
+
+        #
+        # It replaces all following occurrences by 1
+        # which is enough to fix syntax of SQL query
+        # and make sqlglot parser happy:
+        #
+        #   <[Parameters].[SomeParameterName]>
+        #   <Parameters.[SomeParameterName]>
+        #   <[Parameters].SomeParameterName>
+        #   <Parameters.SomeParameterName>
+        #
+        return (
+            re.sub(r"\<\[[Pp]arameters\]\.\[?[^\]]+\]?\>", "1", query)
+            .replace("<<", "<")
+            .replace(">>", ">")
+            .replace("\n\n", "\n")
+        )
```

Comment:
regexes like this scare me - can we add some unit tests for this method? making it a classmethod/staticmethod will also make it easier to test directly

Reply:
done https://github.com/datahub-project/datahub/pull/9838/commits/9be09d303e2e272a6a2ecae9427db4719f9b645d